File: PlantsWeb/src/app/models/ItemDto.ts
import { SafeUrl } from "@angular/platform-browser";
import { ItemType } from "./ItemType";

export interface ItemDto {
        id : string,
        name : string,
        type: ItemType,
        price: number,
        description : string | null,
        imageData : string | null | SafeUrl,
        date: Date | null,
        sold: boolean,
        username: string,
        email: string,
        userId: string    
}

File: PlantsAPI/Models/ItemDto.cs
namespace PlantsAPI.Models
{
    public class ItemDto
    {
        public Guid Id { get; set; }
        public string Name { get; set; }
        public ItemType Type { get; set; }
        public int Price { get; set; }
        public string? Description { get; set; }
        public byte[]? ImageData { get; set; }
        public DateTime Date { get; set; }
        public string Username { get; set; }
        public string Email { get;set; }
        public Guid UserId { get; set; }
        public bool Sold { get; set; }

    }
}
